Source/MQTTnet/Server/MqttClientSessionsManager.cs

```diff
@@ -345,7 +345,7 @@ private async Task<MqttClientConnection> CreateConnectionAsync(IMqttChannelAdapt
 
                 if (session == null)
                 {
-                    session = new MqttClientSession(connectPacket.ClientId, _eventDispatcher, _options, _logger);
+                    session = new MqttClientSession(connectPacket, _eventDispatcher, _options, _logger);
```

Comment:
This place is not enough. The connect packet must be updated also if a session already exists and gets taken over by a new connection (and new values in the CONNECT packet).

My idea was to remove it from the constructor and just making a property with a getter and setter or a method (like _ImportConnectPacketData_) in order to copy the required data (or the while CONNECT packet as you prefer).